website/public/js/controllers/groupsCtrl.js

```diff
@@ -579,17 +578,24 @@ habitrpg.controller("GroupsCtrl", ['$scope', '$rootScope', 'Shared', 'Groups', '
         User.set({'invitations.party':{}});
       }
 
-      $scope.questCancel = function(){
+      $scope.questCancel = function(party){
         if (!confirm(window.env.t('sureCancel'))) return;
-        $rootScope.party.$questCancel();
+        Groups.questCancel(party);
       }
 
-      $scope.questAbort = function(){
+      $scope.questAbort = function(party){
         if (!confirm(window.env.t('sureAbort'))) return;
         if (!confirm(window.env.t('doubleSureAbort'))) return;
-        $rootScope.party.$questAbort();
+        Groups.questAbort(party);
       }
 
+      $scope.questAccept = function(party){
+        Groups.questAccept(party);
```

Comment:
Why not add `User.set({'party.quest.RSVPNeeded': false});` here and at questReject?